internal/overlord/servstate/manager.go

```diff
@@ -477,3 +477,74 @@ func (m *ServiceManager) CheckFailed(name string) {
 		}
 	}
 }
+
+var shutdownTimeout = killWait + 100*time.Millisecond
+
+// Shutdown shuts down the service manager, stopping all running services.
+func (m *ServiceManager) Shutdown() error {
```

Comment:
There seems to be some encapsulation puncturing here which look like a quick shortcut, and could perhaps be avoided. 

We generally have these concepts well organized:

- Manager: manages the underlying business logic of whatever is being taken care of
- Requests: create the task sets to perform specific roles 
- Handlers: take care of task execution itself, triggered by the state machinery

Meanwhile, the daemon, or anywhere else outside, creates the task sets, and injects into changes as desired for its purposes. Multiple task sets can be mixed and matched as appropriate for the task at hand, and the overlord machinery just makes sure it takes place. This pattern should be visible across the overlord and the entire API.

So what we have here is a bit unusual: we have a method in the manager doing the entire dance by itself, even blocking the entire overlord while its tasks execute. That doesn't feel great. Can we avoid it by doing it in a more traditional organization?

Reply:
I'll take a look at reorganizing this -- thanks for the feedback.

Reply:
I've refactored this so the `servstate.StopRunning` function (in request.go) is the thing that creates the task set, and the caller (in the daemon code) calls that and adds it to a task as part of the shutdown sequence. Yeah, it's more consistent this way, and the diff got slightly shorter (mainly due to slightly shorter tests).